Tasks/PowerShellOnTargetMachines2/Utility.ps1

```diff
@@ -0,0 +1,112 @@
+function Parse-TargetMachineNames {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $machineNames
+    )
+
+    Trace-VstsEnteringInvocation $MyInvocation
+    try {
+        # Any verification on the pattern of the target machine name should be done here.
+        $targetMachineNames = $machineNames.Split(',') | Where-Object { if (![string]::IsNullOrEmpty($_)) { Write-Verbose "TargetMachineName: '$_'" ; $_ } };
```

Comment:
You are returning along with port. Is that what you need?

Reply:
no, i will change this as part of later PR's